source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -1053,6 +1066,23 @@ message TransformationEquationInstruction {
 
 }
 
+message TransformationFlattenConditionalBranch {
+
+  // A transformation that takes a selection construct with a header
+  // containing an OpBranchConditional instruction and flattens it.
+
+  // The label id of the header block
+  uint32 header_block_id = 1;
+
+  // A list of pairs mapping instruction descriptors to fresh ids.
+  // Necessary for store, load and function call instructions.
+  repeated InstructionUint32ListPair instruction_to_fresh_ids = 2;
+
+  // A list of fresh ids to be used for instructions that need them, if
+  // they are not in instruction_to_fresh_ids.
+  repeated uint32 overflow_id = 3;
```

Comment:
I wonder if we need this if all ids are accounted for in the previous field.

Reply:
We might need this because shrinking may cause some instruction that would otherwise be removed to remain, requiring additional ids

Reply:
Right, @paulthomson ?

Reply:
I'm not sure but I don't think the shrinker can create additional instructions that weren't present during the fuzzing. But I'm not sure. @paulthomson, what do you think?

Reply:
Even if this is the case, we could probably not apply the transformation at all. This requires some thought, though.

Reply:
A clarification on the last comment: 
Even if the shrinker can create new instructions, I think it would be better not to apply the transformation if we don't have a sufficient number of fresh ids.

P.S. We could have a transformation that changes (for whatever reasons) one memory instruction into another (e.g. OpStore into OpLoad). In this case, if the shrinker removes this transformation from the sequence, we could be lacking fresh ids (because OpLoad requires more of them than OpStore).

Reply:
That is the purpose of the overflow ids: to have fresh ids even when we see "unexpected" instructions so we are still able to apply the transformation. Do you think that we should not apply the transformation in this case? (In your example, there would be no match in the map, since the instruction descriptor for the instruction would not be valid anymore)

Reply:
Yeah, you are right. The transformation might be inapplicable any time one changes some instructions since instruction descriptor depends on the relative order of instructions. That's not the best approach.

Still, it looks odd to me that we have two fields in the message that serve the same purpose (even if it's probably not a big deal).

@stefanomil, what do you think about leaving `overflow_id` and removing the other one?

Reply:
Having a map helps preserve stability of fresh ids: the same instructions will get the same ids even if some previous instructions are removed (while if we just had a list, it would presumably get ids that come sooner in the list), so possible subsequent transformation would still be applicable. But just having a map would make the transformation less flexible, in that some changes could easily make it inapplicable.

So there are two fields so that the transformation is both stable and flexible.

@paulthomson can probably clarify better

Reply:
Indeed. Thanks for the clarification!

In that case, it might be better to mention some points from this discussion in the comments. Something like:
1. for the map field: "This field helps to preserve the stability of the transformation's result since every instruction has a specific set of ids that it can use"
2. for the `overflow_id`: "This field makes the transformation more flexible during the shrinking since every entry (the instruction descriptor) in the previous field depends on the relative position of the instructions in the module".

Nice idea BTW.